service/matching/taskListManager.go

```diff
@@ -236,18 +236,26 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params addTaskParams)
 			return nil, err
 		}
 
+		isForwarded := params.forwardedFrom != ""
+
 		if domainEntry.GetDomainNotActiveErr() != nil {
-			r, err := c.taskWriter.appendTask(params.execution, params.taskInfo)
+			// standby task, only persist when task is not forwarded from child partition
 			syncMatch = false
+			if isForwarded {
```

Comment:
seems like we could also adding code to not try forwarding to root when the domain is not active. (but this is needed anyway, due to racing condition)

Reply:
yes, I will create a separate for not forwarding on standby

Reply:
https://github.com/uber/cadence/issues/4656